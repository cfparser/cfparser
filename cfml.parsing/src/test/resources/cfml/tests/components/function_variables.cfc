component {

	function foo(){
		var foo = function(){
			return "bar!";
		};
	}

}
