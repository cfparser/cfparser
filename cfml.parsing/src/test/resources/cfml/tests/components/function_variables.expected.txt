/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
VAR                      <var>
IDENTIFIER               <foo>
'='                      <=>
FUNCTION                 <function>
'('                      <(>
')'                      <)>
'{'                      <{>
RETURN                   <return>
STRING_LITERAL           <"bar!">
';'                      <;>
'}'                      <}>
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier foo)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (localAssignmentExpression
                var
                (identifier foo)
                =
                (startExpression
                  (compareExpression
                    (baseExpression
                      (anonymousFunctionDeclaration
                        function
                        (
                        parameterList
                        )
                        (compoundStatement
                          {
                          (statement
                            (returnStatement
                              return
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (unaryExpression (primaryExpression (literalExpression "bar!")))
                                  )
                                )
                              )
                              ;
                            )
                          )
                          }
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/